samples/net/mqtt_publisher/prj_pca10056.conf

```diff
@@ -0,0 +1,19 @@
+CONFIG_NETWORKING=y
```

Comment:
As it was pointed out in the issue liked, we rather tend to move into overlay configs instead of creating separate configurations for each board (unless it's really needed).

For instance here's `overlay-bt.conf` I've created and used for tests with nrf52840_pca10056:
```
CONFIG_BT=y
CONFIG_BT_DEBUG_LOG=y
CONFIG_BT_PERIPHERAL=y
CONFIG_BT_L2CAP_DYNAMIC_CHANNEL=y
CONFIG_BT_DEVICE_NAME="Zephyr MQTT"
CONFIG_NET_L2_BT=y
CONFIG_NET_IPV4=n
CONFIG_NET_IPV6=y
CONFIG_NET_CONFIG_MY_IPV4_ADDR=""
CONFIG_NET_CONFIG_PEER_IPV4_ADDR=""
CONFIG_NET_CONFIG_BT_NODE=y
```

With this approach we could move `prj_qemu_x86.conf` into generic `prj.conf` and use it as a base configuration for the sample, and then enable bluetooth with `-DOVERLAY_CONFIG=overlay-bt.conf` option.

If you have time to change this it would be appreciated, otherwise let me know, and I could prepare a separate PR with a cleanup later.

Reply:
> With this approach we could move prj_qemu_x86.conf into generic prj.conf and use it as a base configuration for the sample, and then enable bluetooth with -DOVERLAY_CONFIG=overlay-bt.conf option.

I can do that. One more thing I'll need some help with is using the TLS option, I'm trying it out right now, will ping you if I can't run the sample with TLS.

Reply:
Oh, there is a `overlay-tls.conf`. Is there a way currently to use multiple overlay files?

Reply:
Not sure about it, I'll have to try it out myself :)

Reply:
Hmm, it seems that applyting multiple overlay configs with `-DOVERLAY_CONFG` does not work, unless I did something wrong. But you can always call it like this;
`cmake -DBOARD=nrf52840_pca10056 -DCONF_FILE="prj.conf overlay-bt.conf overlay-tls.conf" ..`

Reply:
I'm happy to tell that TLS with BLE worked out of the box :) We just need to move the TLS initialization out of the loop, as it will fail in a consecutive fail (already registered certs).

Reply:
are the following necessary? Seems to work with default values for me:
```
CONFIG_NET_PKT_RX_COUNT=16
CONFIG_NET_PKT_TX_COUNT=16
CONFIG_NET_BUF_RX_COUNT=16
CONFIG_NET_BUF_TX_COUNT=16
```

Reply:
Not sure about it, aren't defaults larger than that? If so, I'd leave it as it is to save some RAM.